pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -135,16 +135,15 @@ def setup_distributed(self) -> None:
 
     def setup(self, trainer: "pl.Trainer") -> None:
         self.accelerator.setup(trainer)
-        self.setup_optimizers(trainer)
-        self.setup_precision_plugin()
-        optimizers_to_device(self.optimizers, self.root_device)
 
```

Comment:
we were calling `configure_optimizers` 3 times in this one method 😞 
i can split these out as separate PRs if desired

Reply:
Would be better.